src/main/webapp/app/exercises/programming/manage/grading/feedback-analysis/Modal/feedback-filter-modal.component.html

```diff
@@ -0,0 +1,74 @@
+<form [formGroup]="filterForm" (ngSubmit)="applyFilter()">
+    <div class="modal-header">
+        <h5 class="modal-title" jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.modalTitle"></h5>
+        <button type="button" class="btn-close" (click)="closeModal()"></button>
+    </div>
+    <div class="modal-body">
+        <div class="form-group">
+            <label jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.task"></label>
+            <div class="row">
+                @for (task of generateTaskArray(); track task) {
+                    <div class="col-4">
+                        <div class="form-check">
+                            <input
+                                type="checkbox"
+                                class="form-check-input"
+                                id="task-{{ task }}"
+                                [value]="task"
+                                (change)="onCheckboxChange($event, 'tasks')"
+                                [checked]="filterForm.value.tasks.includes(task.toString())"
+                            />
+                            <label class="form-check-label">{{ task }}</label>
+                        </div>
+                    </div>
+                }
+            </div>
+        </div>
+        <div class="form-group">
+            <label jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.testcase"></label>
+            <div class="row">
+                @for (testCase of testCaseNames(); track testCase) {
+                    <div class="col-4">
+                        <div class="form-check">
+                            <input
+                                type="checkbox"
+                                class="form-check-input"
+                                id="testCase-{{ testCase }}"
+                                [value]="testCase"
+                                (change)="onCheckboxChange($event, 'testCases')"
+                                [checked]="filterForm.value.testCases.includes(testCase)"
+                            />
+                            <label class="form-check-label">{{ testCase }}</label>
+                        </div>
+                    </div>
+                }
+            </div>
+        </div>
+        <label jhiTranslate="artemisApp.programmingExercise.configureGrading.feedbackAnalysis.filterModal.occurrence"></label>
+        <jhi-range-slider
+            [generalMinValue]="minCount()"
+            [generalMaxValue]="maxCount()"
+            [(selectedMinValue)]="occurrence[0]"
+            [(selectedMaxValue)]="occurrence[1]"
```

Comment:
`occurrence` is also technically a function call (get/set pair). Could we circumvent this?

Reply:
What’s the status here?

Reply:
done now 👍